### PR TITLE
Add ONNX Round op

### PR DESCRIPTION
**Pull request by mbencer**
Round op: https://github.com/onnx/onnx/blob/master/docs/Operators.md#Round
WARNING: There are some cases when result can be incorrect
The full support solution (now blocked) is in Draft: https://github.com/openvinotoolkit/openvino/pull/1676

### CODE REVIEW

File: ngraph/src/ngraph/frontend/onnx_import/op/round.cpp [hunk 1]
```diff
@@ -30,10 +30,44 @@ namespace ngraph
...
-                    return {std::make_shared<ngraph::opset0::Round>(data)};
+
+                    Output<ngraph::Node> one_const =
+                        default_opset::Constant::create(element::f32, {}, {1.0f});
+                    Output<ngraph::Node> half_const =
+                        default_opset::Constant::create(element::f32, {}, {0.5f});
+                    if (data.get_element_type() != element::f32)
+                    {
+                        one_const = std::make_shared<default_opset::Convert>(
+                            one_const, data.get_element_type());
```
Comment (tsocha) [inline ngraph/src/ngraph/frontend/onnx_import/op/round.cpp:55]: `one_const` can be created with a proper data type instead of converting it into it.

File: ngraph/src/ngraph/frontend/onnx_import/op/round.cpp [hunk 2]
```diff
@@ -30,10 +30,44 @@ namespace ngraph
...
+                        half_const = std::make_shared<default_opset::Convert>(
+                            half_const, data.get_element_type());
+                    }
+
+                    const auto data_floor = std::make_shared<default_opset::Floor>(data);
+                    const auto data_floor_plus_one =
+                        std::make_shared<default_opset::Add>(data_floor, one_const);
+
+                    const auto diff = std::make_shared<default_opset::Add>(
+                        data, std::make_shared<default_opset::Negative>(data_floor));
```
Comment (tsocha) [inline ngraph/src/ngraph/frontend/onnx_import/op/round.cpp:65]: Why are you adding the negative `data_floor` instead of simple subtraction?

File: ngraph/src/ngraph/frontend/onnx_import/op/round.cpp (hunk 1 above)
Comment (mbencer) [reply ngraph/src/ngraph/frontend/onnx_import/op/round.cpp]: done

File: ngraph/src/ngraph/frontend/onnx_import/op/round.cpp (hunk 2 above)
Comment (mbencer) [reply ngraph/src/ngraph/frontend/onnx_import/op/round.cpp]: right ;)

File: ngraph/test/onnx/onnx_import.in.cpp [hunk 3]
```diff
@@ -2262,23 +2262,23 @@ NGRAPH_TEST(${BACKEND_NAME}, onnx_model_round)
...
+}
+
+// CASES NOT CORRECTLY HANDLED BY CURRENT IMPLEMENTATION OF ROUND
+NGRAPH_TEST(${BACKEND_NAME}, onnx_model_round_half_nearest_even)
+{
+    const auto function = onnx_import::import_onnx_model(
+        file_util::path_join(SERIALIZED_ZOO, "onnx/round_half_nearest_even.prototxt"));
+    auto test_case = test::TestCase<TestEngine>(function);
+
+    test_case.add_input<float>({0.5f, 2.5f, -1.5f - 2.5f});
```
Comment (mitruska) [inline ngraph/test/onnx/onnx_import.in.cpp:2280]:
```suggestion
    test_case.add_input<float>({0.5f, 2.5f, -1.5f, -2.5f});
```

File: ngraph/src/ngraph/frontend/onnx_import/op/round.cpp [hunk 4]
```diff
@@ -30,10 +30,36 @@ namespace ngraph
...
+                // Current version is:
+                // data_floor = floor(data)
+                // diff = data - data_floor
+                // if(diff < 0.5f)
+                //   return data_floor
+                // else
+                //   return data_floor + 1.0f
+                //
+                // The correct version should contain condition:
+                // if (diff < 0.5f || (diff == 0.5f && static_cast<int>(data_floor) % 2 == 0))
```
Comment (mitruska) [inline ngraph/src/ngraph/frontend/onnx_import/op/round.cpp:43]: I just wonder, why we can't implement this additional condition with the Mod operator?

Comment (mbencer) [reply ngraph/src/ngraph/frontend/onnx_import/op/round.cpp:43]: @mitruska I've tried to do it, but it fails during decomposition of `Mod` operator (`Mod ` is fused operator) - because of not known shape during decomposition.
What's more the produced subgraph was pretty complicated for rather simple operation.
But I can check it once more (maybe it is possible to extend shape propagation for this case)

Comment (mitruska) [reply ngraph/src/ngraph/frontend/onnx_import/op/round.cpp:43]: Sure, I understand

File: ngraph/frontend/onnx_import/src/onnx_import/op/round.cpp [hunk 5]
```diff
@@ -30,10 +30,36 @@ namespace ngraph
         {
             namespace set_1
             {
+                // WARNING!
```
Comment (lazarevevgeny) [inline ngraph/frontend/onnx_import/src/onnx_import/op/round.cpp:33]: I think that `Round(x) = Floor(x + 0.5 * Sign(x))`. It is a bit easier to implement and should work faster (because all the operations are elementwise and should be effectively fused in the plugin) than what you wrote below. It also works for negative numbers. Please, correct me if I am wrong.

Comment (mbencer) [reply ngraph/frontend/onnx_import/src/onnx_import/op/round.cpp:33]: @lazarevevgeny 
Let's consider `data=-1.1 -> Floor(-1.1 + 0.5*(-1)) = Floor(-1.6) = -2`
while for current version:
`data_floor = Floor(-1.1) = -2`
`diff = data - data_floor = -1.1 - (-2) = 0.9`
because (diff < 0.5) != true
`round = -2 + 1 = -1`
So I am not sure about negative numbers. Do I miss something?

Comment (lazarevevgeny) [reply ngraph/frontend/onnx_import/src/onnx_import/op/round.cpp]: Ok. I see the problem in my formula. How about this one: `Round(x) = Floor(x + 0.5)`?

Comment (mbencer) [reply ngraph/frontend/onnx_import/src/onnx_import/op/round.cpp]: @lazarevevgeny Good point, it works correctly for all cases supported by current version. Thanks

File: ngraph/test/runtime/ie/unit_test.manifest [hunk 6]
```diff
@@ -47,8 +47,8 @@ onnx_model_scatterND
 onnx_model_gatherND_int32
 onnx_model_gatherND_float
 
-# Not supported ONNX op: Round
-onnx_model_round
+# Round op doesn't some specific cases of rounding
```
Comment (postrational) [inline ngraph/test/runtime/ie/unit_test.manifest:50]:
```suggestion
# Round op doesn't support some specific cases of rounding
```

File: ngraph/test/runtime/interpreter/unit_test.manifest [hunk 7]
```diff
@@ -110,4 +110,6 @@ INTERPRETER.onnx_model_conv_integer_pads
 INTERPRETER.onnx_model_scatterND
 INTERPRETER.onnx_model_gatherND_int32
 INTERPRETER.onnx_model_gatherND_float
-INTERPRETER.onnx_model_round
+
+# Round op doesn't some specific cases of rounding
```
Comment (postrational) [inline ngraph/test/runtime/interpreter/unit_test.manifest:114]:
```suggestion
# Round op doesn't support some specific cases of rounding
```

File: ngraph/python/tests/test_onnx/test_zoo_models.py [hunk 8]
```diff
@@ -616,6 +616,7 @@
 
 xfail_issue_36537(test_cases.test_shufflenet_v2_opset10_cpu)
 xfail_issue_36538(test_cases.test_yolov3_opset10_cpu)
+xfail_issue_36538(test_cases.test_tiny_yolov3_opset11_cpu)
```
Comment (postrational) [inline ngraph/python/tests/test_onnx/test_zoo_models.py:619]: Why did you need to add this xfail?

Comment (mbencer) [reply ngraph/python/tests/test_onnx/test_zoo_models.py:619]: @postrational  After enabling `Round` the next problem occurs. It is the same exception as for `yolov3_opset10_cpu` - there is created bug for it.
